smart-contracts/contracts/cl-vault/src/vault/deposit.rs

```diff
@@ -46,3 +68,523 @@ pub(crate) fn execute_deposit(
     // // Merge responses and add message to mint vault token
     // Ok(merge_responses(vec![receive_res, compound_res, mint_res]).add_event(event))
 }
+
+pub(crate) fn execute_multi_deposit(
+    deps: DepsMut,
+    env: Env,
+    info: &MessageInfo,
+    recipient: Option<String>,
+) -> Result<Response, ContractError> {
+    // Unwrap recipient or use caller's address
+    let recipient = recipient.map_or(Ok(info.sender.clone()), |x| deps.api.addr_validate(&x))?;
+
+    let pool = POOL_CONFIG.load(deps.storage)?;
+    let (token0, token1) = must_pay_two(&info, (pool.token0, pool.token1))?;
+
+    let position = POSITION.load(deps.storage)?;
+    let range = ConcentratedliquidityQuerier::new(&deps.querier)
+        .position_by_id(position.position_id)?
+        .position
+        .ok_or(ContractError::PositionNotFound)?
+        .position
+        .ok_or(ContractError::PositionNotFound)?;
+
+    let create_msg = create_position(
+        deps.storage,
+        &env,
+        range.lower_tick,
+        range.upper_tick,
+        vec![token0.clone(), token1.clone()],
+        Uint128::zero(),
+        Uint128::zero(),
+    )?;
```

Comment:
In case of lopsided input, do we refund all back to depositor? Or should we execute a sdm after (swap-deposit-merge)

Reply:
Currently we refund all, I think it makes sense to offer this on it's own, we can hookup a version with sdm to single sided, maybe make that accept multiple assets and just figure it out